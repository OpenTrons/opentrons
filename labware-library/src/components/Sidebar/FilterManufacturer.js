// @flow
// filter labware by manufacturer
import * as React from 'react'
import { withRouter } from 'react-router-dom'

import { SelectField } from '@opentrons/components'
import { getAllManufacturers, buildFiltersUrl } from '../../filters'
import styles from './styles.css'

import type { ContextRouter } from 'react-router-dom'
import type { FilterParams } from '../../types'

// TODO(mc, 2019-03-13): i18n
const EN_MANUFACTURER = 'manufacturer'

// TODO(mc, 2019-03-18): i18n
const EN_MANUFACTURER_LABELS = {
  all: 'All',
  generic: 'Generic',
}

export type FilterManufacturerProps = {
  ...ContextRouter,
  filters: FilterParams,
}

export function FilterManufacturer(props: FilterManufacturerProps) {
  const { history, filters } = props
  const manufacturers = getAllManufacturers()
  const options = manufacturers.map(value => ({
    value,
    label: EN_MANUFACTURER_LABELS[value] || value,
  }))

  return (
    <label className={styles.filter_manufacturer}>
      <p className={styles.filter_manufacturer_label}>{EN_MANUFACTURER}</p>
      <SelectField
        className={styles.filter_manufacturer_select}
        name="manufacturer"
        value={filters.manufacturer}
        options={options}
        onValueChange={(_, value) => {
          if (value) {
            history.push(buildFiltersUrl({ ...filters, manufacturer: value }))
          }
        }}
      />
    </label>
  )
}

export default withRouter(FilterManufacturer)
