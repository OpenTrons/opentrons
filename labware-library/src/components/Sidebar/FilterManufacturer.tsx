// filter labware by manufacturer
import { getAllManufacturers, buildFiltersUrl } from '../../filters'
import { MANUFACTURER, MANUFACTURER_VALUES } from '../../localization'
import type { FilterParams } from '../../types'
import styles from './styles.css'
import { SelectField } from '@opentrons/components'
import type { SelectOptionOrGroup } from '@opentrons/components'
import * as React from 'react'
import { withRouter } from 'react-router-dom'
import type { RouteComponentProps } from 'react-router-dom'

export interface FilterManufacturerProps extends RouteComponentProps {
  filters: FilterParams
}

export function FilterManufacturerComponent(
  props: FilterManufacturerProps
): JSX.Element {
  const { history, filters } = props
  const manufacturers = getAllManufacturers()
  const options: SelectOptionOrGroup[] = manufacturers.map(value => ({
    value,
    label: MANUFACTURER_VALUES[value] || value,
  }))

  options.push({ options: [{ label: 'Other', value: 'other' }] })

  return (
    <label className={styles.filter_manufacturer}>
      <p className={styles.filter_label}>{MANUFACTURER}</p>
      <SelectField
        className={styles.filter_manufacturer_select}
        name="manufacturer"
        value={filters.manufacturer}
        options={options}
        onValueChange={(_, value) => {
          if (value) {
            history.push(buildFiltersUrl({ ...filters, manufacturer: value }))
          }
        }}
      />
    </label>
  )
}

export const FilterManufacturer = withRouter(FilterManufacturerComponent)
