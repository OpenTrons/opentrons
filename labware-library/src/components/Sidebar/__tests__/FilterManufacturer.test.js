// @flow
// FilterManufacturer component tests
import * as React from 'react'
import Renderer from 'react-test-renderer'

import FilterManufacturer from '../FilterManufacturer'

describe('FilterManufacturer', () => {
  test('component renders', () => {
    const tree = Renderer.create(<FilterManufacturer />).toJSON()

    expect(tree).toMatchSnapshot()
  })

  // TODO(mc, 2019-03-14): test manufacturer list population
})
