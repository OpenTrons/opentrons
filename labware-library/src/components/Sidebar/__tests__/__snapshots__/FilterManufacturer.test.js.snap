// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterManufacturer component renders 1`] = `
<label
  className="filter_manufacturer"
>
  <p
    className="filter_manufacturer_label"
  >
    manufacturer
  </p>
  <div>
    <div
      className="css-10nd86i filter_manufacturer_select"
      onKeyDown={[Function]}
    >
      <div
        className="css-0 select_control"
        onMouseDown={[Function]}
        onTouchEnd={[Function]}
      >
        <div
          className="css-1hwfws3"
        >
          <div
            className="css-xp4uvy"
          >
            All
          </div>
          <div
            className="css-y8aj3r"
          >
            <div
              className=""
              style={
                Object {
                  "display": "inline-block",
                }
              }
            >
              <input
                aria-autocomplete="list"
                autoCapitalize="none"
                autoComplete="off"
                autoCorrect="off"
                disabled={false}
                id="react-select-2-input"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                spellCheck="false"
                style={
                  Object {
                    "background": 0,
                    "border": 0,
                    "boxSizing": "content-box",
                    "color": "inherit",
                    "fontSize": "inherit",
                    "opacity": 1,
                    "outline": 0,
                    "padding": 0,
                    "width": "1px",
                  }
                }
                tabIndex="0"
                type="text"
                value=""
              />
              <div
                style={
                  Object {
                    "height": 0,
                    "left": 0,
                    "overflow": "scroll",
                    "position": "absolute",
                    "top": 0,
                    "visibility": "hidden",
                    "whiteSpace": "pre",
                  }
                }
              >
                
              </div>
            </div>
          </div>
        </div>
        <div
          className="css-1wy0on6"
        >
          <div
            aria-hidden="true"
            className="css-1ep9fjw"
            onMouseDown={[Function]}
            onTouchEnd={[Function]}
          >
            <div
              className="dropdown_icon_wrapper"
            >
              <svg
                aria-hidden="true"
                className=""
                fill="currentColor"
                version="1.1"
                viewBox="0 0 24 24"
                width="100%"
              >
                <path
                  d="M7,10L12,15L17,10H7Z"
                  fillRule="evenodd"
                />
              </svg>
            </div>
          </div>
        </div>
      </div>
      <input
        name="manufacturer"
        type="hidden"
        value="all"
      />
    </div>
  </div>
</label>
`;
